wrapanapi/systems/openstack.py

```diff
@@ -1059,3 +1068,92 @@ def usage_and_quota(self):
             'cpu_total': host_cpus,
             'cpu_limit': data['maxTotalCores'] if data['maxTotalCores'] >= 0 else None,
         }
+
+    def list_containers(self):
+        """List of available containers.
+
+        Returns: list of containers name.
+        """
+
+        _, containers = self.sapi.get_account()
+        return [cont.get("name") for cont in containers]
+
+    def create_container(self, container_name):
+        """Create container.
+
+        Args:
+            container_name: name of container
+
+        Return: None
+        """
+
+        self.sapi.put_container(container_name)
+
+    def delete_container(self, container_name):
+        """Delete existing container
+
+        Args:
+            container_name: name of container
+
+        Returns: bool
+        """
+
+        self.sapi.delete_container(container_name)
+        return container_name not in self.list_containers()
+
+    def list_objects(self, container_name):
+        """List of available object in container.
+
+        Args:
+            container_name: name of container
+
+        Returns: list of existing objects
+        """
+
+        _, objects = self.sapi.get_container(container_name)
+        return [obj.get("name") for obj in objects]
+
+    def create_object(self, container_name, path, object_name=None):
+        """Upload the object under container.
+
+        Args:
+            container_name: name of container
+            path: local object path
+            object_name: set name of object different than local.
+
+        Return: None
+        """
+
+        name = object_name or os.path.basename(path)
+
+        with open(path, 'rb') as obj:
+            self.sapi.put_object(container_name, name, contents=obj)
+
+    def delete_object(self, container_name, object_name):
+        """Delete object from container.
+
+        Args:
+            container_name: name of container
+            object_name: name of object
+
+        Returns: bool
+        """
+
+        self.sapi.delete_object(container_name, object_name)
```

Comment:
Again, handling of ClientException when the container, or object, doesn't exist.